core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -293,13 +294,26 @@ public PlanAndMappings visitTableScan(TableScanNode node, UnaliasContext context
 
             List<Symbol> newOutputs = mapper.map(node.getOutputSymbols());
 
+            Optional<PlanNodeStatsEstimate> statistics = node.getStatistics();
+            PlanNodeStatsEstimate.Builder newStatistics = PlanNodeStatsEstimate.builder();
+            statistics.ifPresent(stats -> newStatistics.setOutputRowCount(stats.getOutputRowCount()));
             Map<Symbol, ColumnHandle> newAssignments = new HashMap<>();
             node.getAssignments().forEach((symbol, handle) -> {
-                newAssignments.put(mapper.map(symbol), handle);
+                Symbol newSymbol = mapper.map(symbol);
+                newAssignments.put(newSymbol, handle);
+                statistics.ifPresent(stats -> newStatistics.addSymbolStatistics(newSymbol, stats.getSymbolStatistics(symbol)));
             });
 
             return new PlanAndMappings(
-                    new TableScanNode(node.getId(), node.getTable(), newOutputs, newAssignments, node.getEnforcedConstraint(), node.isUpdateTarget(), node.getUseConnectorNodePartitioning()),
+                    new TableScanNode(
+                            node.getId(),
+                            node.getTable(),
+                            newOutputs,
+                            newAssignments,
+                            node.getEnforcedConstraint(),
+                            statistics.isPresent() ? Optional.of(newStatistics.build()) : Optional.empty(),
```

Comment:
you can do `the `? :` part internally in the constructor.

Reply:
i don't understand

Reply:
never mind; misread